docs/spec-v1alpha1.md

```diff
@@ -0,0 +1,221 @@
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**  *generated with [DocToc](https://github.com/thlorenz/doctoc)*
+
+- [OCI Model Configuration Specification](#oci-model-configuration-specification)
+  - [Terminology](#terminology)
+    - [Tensor](#tensor)
+    - [GitRepo](#gitrepo)
+  - [Properties](#properties)
+  - [Example](#example)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+# OCI Model Configuration Specification
+
+An OCI Model is a Machine Learning/Deep Learning model. This specification outlines the JSON format describing models for use and execution tool and its relationship to filesystem changesets.
+
+This section defines the `application/vnd.caicloud.model.config.v1alpha1+json` media type.
+
+## Terminology
+
+This specification uses the following terms:
+
+### [Tensor](https://www.tensorflow.org/guide/tensor)
+
+Tensors are multi-dimensional arrays with a uniform type.
+
+- **name** string, OPTIONAL
+
+    Name of the tensor.
+
+- **description** string, OPTIONAL
+
+    Description of the tensor.
+
+- **dtype** string, OPTIONAL
+
+    DType of the tensor.
+
+- **size** object, OPTIONAL
+
+    Size of the tensor.
+
+### [GitRepo](https://kubernetes.io/docs/concepts/storage/volumes/#gitrepo)
+
+GitRepo is the Git repository.
+
+- **repository** string, OPTIONAL
+
+    Repository of the training scripts
+
+- **revision** string, OPTIONAL
+
+    Revision of the repository
+
+- **description** string, OPTIONAL
+
+    Description of the training script
+
+## Properties
+
+Note: Any OPTIONAL field MAY also be set to null, which is equivalent to being absent.
+
+- **created** string, OPTIONAL
+
+    A combined date and time at which the image was created, formatted as defined by RFC 3339, section 5.6.
+
+- **author** string, OPTIONAL
+
+    Gives the name and/or email address of the person or entity which created and is responsible for maintaining the image.
+
+- **description** string, OPTIONAL
+
+    Description of the model
+
+- **tags** object, OPTIONAL
+
+    Tags of the model
+
+- **labels** object, OPTIONAL
+
+    The field contains arbitrary metadata for the container. This property MUST use the annotation rules.
+
+- **framework** string, OPTIONAL
+
+    The framework of the model (e.g. TensorFlow)
+
+- **format** string, REQUIRED
+
+    The format of the model (savedmodel, onnx and so on)
+
+- **size** int, REQUIRED
```

Comment:
Is there any unit requirement for `size`? e.g. `10Gi` or `10G` or others?

Reply:
The size is in defined in containerd github.com/containerd/containerd/content/content.go

We convert it to human-readable size:

```
// byteCountBinary produces a human-readable file size
func byteCountBinary(b int64) string {
	const unit = 1024
	if b < unit {
		return fmt.Sprintf("%d B", b)
	}
	div, exp := int64(unit), 0
	for n := b / unit; n >= unit; n /= unit {
		div *= unit
		exp++
	}
	return fmt.Sprintf("%.1f %ciB", float64(b)/float64(div), "KMGTPE"[exp])
}
```

Do you think we should keep a string to use the human-readable representation?

Reply:
what do you mean by ` keep a string to use the human-readable representation?`

Reply:
Now I propose to use `**size** int, REQUIRED`, do you want to use `**size** string, REQUIRED` to store `8Gi` or `8Mi`?

Reply:
If this is mostly generated automatically, then using `int` would be fine.